base/show.jl

```diff
@@ -1483,15 +1485,18 @@ function show_unquoted(io::IO, ex::Expr, indent::Int, prec::Int, quote_level::In
         arg1 = args[1]
         if arg1 isa Symbol
             show_sym(io, arg1, allow_macroname=true)
+        elseif arg1 isa GlobalRef
+            show_globalref(io, arg1, allow_macroname=true)
         elseif is_expr(arg1, :(.)) && length(arg1.args) == 2
-            if arg1.args[1] isa Module
-                print(io, '(')
-                print(io, arg1.args[1])
-                print(io, ").")
+            m = arg1.args[1]
+            if m isa Symbol || m isa GlobalRef || is_expr(m, :(.), 2)
```

Comment:
Very minor: should we also allow `m isa Module` here in case people interpolate a module into the AST?

Reply:
The previous version of the code parenthesized Modules, but I'm not sure why. I guess we'd need `$` to be super accurate about it.

Reply:
Good point, it would be nice to generally show the difference between purely symbolic ASTs vs ASTs which have "unusual" objects interpolated into them. Especially for things like modules which `show` the same way as symbols.